docs/operator-manual/high_availability.md

```diff
@@ -60,15 +60,14 @@ reconciliation. In this case, we advise to use the preferred resource version in
 * The controller polls Git every 3m by default. You can change this duration using the `timeout.reconciliation` setting in the `argocd-cm` ConfigMap. The value of `timeout.reconciliation` is a duration string e.g `60s`, `1m`, `1h` or `1d`.
 
 * If the controller is managing too many clusters and uses too much memory then you can shard clusters across multiple
-controller replicas. To enable sharding increase the number of replicas in `argocd-application-controller` `StatefulSet`
-and repeat the number of replicas in the `ARGOCD_CONTROLLER_REPLICAS` environment variable. The strategic merge patch below
+controller replicas. To enable sharding increase the number of replicas in `argocd-application-controller` `Deployment`. The strategic merge patch below
```

Comment:
I would be great to have something in the docs describing how the dynamic shard rebalance works. Things like:
- How to enable it?
- What is the expected behaviour?
- What are the different methods of rebalancing shards implemented in Argo CD today?
- What is the guideline to switch the controller from StatefulSet to Deployment?

Reply:
> * How to enable it?
It would be enabled by default when users upgrade to v2.9, as it will deploy application controller as a deployment instead of a statefulset.

> What is the expected behaviour?
I added some detail around it.

> What are the different methods of rebalancing shards implemented in Argo CD today?
I think this would be not needed as we already have sharding documented under the argocd-application-controller section [here](https://argo-cd.readthedocs.io/en/stable/operator-manual/high_availability/#argocd-application-controller)

> What is the guideline to switch the controller from StatefulSet to Deployment?
IMO, it would be more useful to have this as part of upgrade notes instead of having it here. WDYT?